axom 0.4.0: build fails: ld: cannot find -lopenmp

`axom@0.4.0` fails to build using:
* spack@develop (087110bcb013566f6ba392d4c271e891f4b3a2b1 from `Thu Apr 29 16:43:01 2021 +0200`)
* Ubuntu 20.04 - GCC 9.3.0
* RHEL 8 - GCC 8.3.1

Using container: `ecpe4s/ubuntu20.04-runner-x86_64:2021-03-10`
Concrete spec: [axom-ayoawp.spec.yaml.txt](https://github.com/spack/spack/files/6400068/axom-ayoawp.spec.yaml.txt)
Build log: [axom-build-out.txt](https://github.com/spack/spack/files/6400078/axom-build-out.txt)

```
$> spack mirror add E4S https://cache.e4s.io
$> spack buildcache keys -it

$> spack install --cache-only --only dependencies --include-build-deps -f ./axom-ayoawp.spec.yaml
... OK

$> spack install --no-cache -f ./axom-ayoawp.spec.yaml
...
==> Installing axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx
==> Warning: Missing a source id for conduit@develop
==> No patches needed for axom
==> axom: Executing phase: 'initconfig'
==> axom: Executing phase: 'cmake'
==> axom: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

4 errors found in build log:
     500          |   ^~~~~~
     501    [ 27%] Linking CXX executable ../../../examples/core_utilities_ex
     502    cd /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/axom/core/examples && /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cma
            ke-3.19.7-7zkgd4xkg62fl5x2upq4mof5dkkkg3u4/bin/cmake -E cmake_link_script CMakeFiles/core_utilities_ex.dir/link.txt --verbose=1
     503    /opt/spack/lib/spack/env/gcc/g++      -Wall -Wextra  -O2 -g -DNDEBUG -Wl,--export-dynamic -rdynamic -fopenmp -Wl,-rpath -Wl,/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mp
            ich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/lib -L/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.4.1-6lkpnwfflqhxirx554hzmbkck4v6vs25/lib CMakeFiles/core_utilities_ex
            .dir/core_utilities.cpp.o ../CMakeFiles/core.dir/utilities/FileUtilities.cpp.o ../CMakeFiles/core.dir/utilities/StringUtilities.cpp.o ../CMakeFiles/core.dir/utilities/Utilities.cpp.o
             ../CMakeFiles/core.dir/utilities/About.cpp.o ../CMakeFiles/core.dir/utilities/nvtx/interface.cpp.o ../CMakeFiles/core.dir/utilities/nvtx/Range.cpp.o ../CMakeFiles/core.dir/numerics/
            polynomial_solvers.cpp.o ../CMakeFiles/core.dir/Types.cpp.o -o ../../../examples/core_utilities_ex  -Wl,-rpath,/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/umpire-4.1.2-of
            zdoemhytkqvaq4vaeifnarrdhfo6ep/lib:/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.13.0-l22sfjknjzhqf6vxsr34apzqkmz3zixd/lib:/opt/spack/opt/spack/linux-ubuntu20.04-x86
            _64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/lib /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/umpire-4.1.2-ofzdoemhytkqvaq4vaeifnarrdhfo6ep/lib/libumpire.so
            /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.13.0-l22sfjknjzhqf6vxsr34apzqkmz3zixd/lib/libRAJA.so /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.
            1-hm77n22t37spis2wa4wssqtmqnvuhfz6/lib/libmpi.so /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/lib/libmpicxx.so /opt/spack/opt/
            spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/lib/libmpifort.so -lopenmp -ldl
     504    [ 27%] Building CXX object axom/mint/CMakeFiles/mint.dir/mesh/blueprint.cpp.o
     505    cd /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/axom/mint && /opt/spack/lib/spack/env/gcc/g++ -DAXOM_DEBUG -I/tmp/root/spack-stag
            e/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/include -I/tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-src/src -I
            /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/camp-0.1.0-cmr4rmbw36urn2b72ghkesab3qepb3iz/include -I/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.13.0-l22s
            fjknjzhqf6vxsr34apzqkmz3zixd/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/umpire-4.1.2-ofzdoemhytkqvaq4vaeifnarrdhfo6ep/include -isystem /opt/spack/opt/sp
            ack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/include -Wall -Wextra  -O2 -g -DNDEBUG -fPIC -fopenmp -std=c++14 -o CMakeFiles/mint.dir/mesh/bluep
            rint.cpp.o -c /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-src/src/axom/mint/mesh/blueprint.cpp
  >> 506    /usr/bin/ld: cannot find -lopenmp
  >> 507    collect2: error: ld returned 1 exit status
  >> 508    make[2]: *** [axom/core/examples/CMakeFiles/core_utilities_ex.dir/build.make:127: examples/core_utilities_ex] Error 1
     509    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa'
  >> 510    make[1]: *** [CMakeFiles/Makefile2:929: axom/core/examples/CMakeFiles/core_utilities_ex.dir/all] Error 2
     511    make[1]: *** Waiting for unfinished jobs....
     512    [ 28%] Building CXX object axom/slam/examples/tinyHydro/CMakeFiles/slam_tinyHydro_ex.dir/PolygonMeshXY.cpp.o
     513    cd /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/axom/slam/examples/tinyHydro && /opt/spack/lib/spack/env/gcc/g++ -DAXOM_DEBUG -I/
            tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/include -I/tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppcey
            rx/spack-src/src -I/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/camp-0.1.0-cmr4rmbw36urn2b72ghkesab3qepb3iz/include -I/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3
            .0/raja-0.13.0-l22sfjknjzhqf6vxsr34apzqkmz3zixd/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/umpire-4.1.2-ofzdoemhytkqvaq4vaeifnarrdhfo6ep/include -isyste
            m /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/include -isystem /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4
            spywmjvpppceyrx/spack-src/src/thirdparty -Wall -Wextra  -O2 -g -DNDEBUG -fPIC -fopenmp -std=c++14 -o CMakeFiles/slam_tinyHydro_ex.dir/PolygonMeshXY.cpp.o -c /tmp/root/spack-stage/spa
            ck-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-src/src/axom/slam/examples/tinyHydro/PolygonMeshXY.cpp
     514    [ 29%] Building CXX object axom/mint/CMakeFiles/mint.dir/mesh/CurvilinearMesh.cpp.o
     515    cd /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/axom/mint && /opt/spack/lib/spack/env/gcc/g++ -DAXOM_DEBUG -I/tmp/root/spack-stag
            e/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa/include -I/tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-src/src -I
            /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/camp-0.1.0-cmr4rmbw36urn2b72ghkesab3qepb3iz/include -I/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.13.0-l22s
            fjknjzhqf6vxsr34apzqkmz3zixd/include -isystem /opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/umpire-4.1.2-ofzdoemhytkqvaq4vaeifnarrdhfo6ep/include -isystem /opt/spack/opt/sp
            ack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.1-hm77n22t37spis2wa4wssqtmqnvuhfz6/include -Wall -Wextra  -O2 -g -DNDEBUG -fPIC -fopenmp -std=c++14 -o CMakeFiles/mint.dir/mesh/Curvi
            linearMesh.cpp.o -c /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-src/src/axom/mint/mesh/CurvilinearMesh.cpp
     516    [ 30%] Building CXX object axom/sidre/CMakeFiles/sidre.dir/core/View.cpp.o

See build log for details:
  /tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-out.txt
```

@white238

> I think i fixed it in the PR mentioned above. Can you see if this fixes your problem?

Thank you for your quick work. Unfortunately, this didn't resolve the issue. I get the same error :(
```
  >> 440    /usr/bin/ld: cannot find -lopenmp
  >> 441    collect2: error: ld returned 1 exit status
  >> 442    make[2]: *** [axom/core/examples/CMakeFiles/core_utilities_ex.dir/build.make:127: examples/core_utilities_ex] Error 1
     443    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-axom-0.4.0-ayoawpaledknbjhc4spywmjvpppceyrx/spack-build-ayoawpa'
  >> 444    make[1]: *** [CMakeFiles/Makefile2:929: axom/core/examples/CMakeFiles/core_utilities_ex.dir/all] Error 2
```

--- a/var/spack/repos/builtin/packages/blt/package.py
+++ b/var/spack/repos/builtin/packages/blt/package.py
@@ -24,7 +24,8 @@ class Blt(Package):
     #  https://llnl-blt.readthedocs.io/en/develop/tutorial/exporting_targets.html
     version('0.4.0', sha256='f3bc45d28b9b2eb6df43b75d4f6f89a1557d73d012da7b75bac1be0574767193')
 
-    # 0.3.6 + a specific fix for rocm, that is before 0.4.0's breaking change
+    # 0.3.6 + a specific fix for versions of packages that rely on rocm and
+    # cannot move completely forward in BLT to the breaking change in 0.4.0
     version('0.3.6rocm', commit='a98a5836c7a9e34b75a54eb46212812796fe5557')
 
     version('0.3.6', sha256='6276317c29e7ff8524fbea47d9288ddb40ac06e9f9da5e878bf9011e2c99bf71')

--- a/var/spack/repos/builtin/packages/blt/package.py
+++ b/var/spack/repos/builtin/packages/blt/package.py
@@ -18,7 +18,15 @@ class Blt(Package):
 
     version('develop', branch='develop')
     version('main', branch='main')
+    # Note: 0.4.0 contains a breaking change to BLT created targets
+    #  if you export targets this could cause problems in downstream
+    #  projects if not handled properly. More info here:
+    #  https://llnl-blt.readthedocs.io/en/develop/tutorial/exporting_targets.html
     version('0.4.0', sha256='f3bc45d28b9b2eb6df43b75d4f6f89a1557d73d012da7b75bac1be0574767193')
+
+    # 0.3.6 + a specific fix for rocm, that is before 0.4.0's breaking change
+    version('0.3.6rocm', commit='a98a5836c7a9e34b75a54eb46212812796fe5557')
+
     version('0.3.6', sha256='6276317c29e7ff8524fbea47d9288ddb40ac06e9f9da5e878bf9011e2c99bf71')
     version('0.3.5', sha256='68a1c224bb9203461ae6f5ab0ff3c50b4a58dcce6c2d2799489a1811f425fb84')
     version('0.3.0', sha256='bb917a67cb7335d6721c997ba9c5dca70506006d7bba5e0e50033dd0836481a5')

--- a/var/spack/repos/builtin/packages/camp/package.py
+++ b/var/spack/repos/builtin/packages/camp/package.py
@@ -22,15 +22,12 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
     # TODO: figure out gtest dependency and then set this default True.
     variant('tests', default=False, description='Build tests')
 
-    depends_on('blt', type='build')
-    depends_on('blt@0.3.7:', type='build', when='+rocm')
     depends_on('cub', when='+cuda')
 
     def cmake_args(self):
         spec = self.spec
 
         options = []
-        options.append('-DBLT_SOURCE_DIR={0}'.format(spec['blt'].prefix))
 
         if '+cuda' in spec:
             options.extend([

--- a/var/spack/repos/builtin/packages/care/package.py
+++ b/var/spack/repos/builtin/packages/care/package.py
@@ -28,7 +28,7 @@ class Care(CMakePackage, CudaPackage, ROCmPackage):
     variant('tests', default=False, description='Build tests')
     variant('loop_fuser', default=False, description='Enable loop fusion capability')
 
-    depends_on('blt', type='build')
+    depends_on('blt', type='build', when='care@0.3.1:')
     depends_on('blt@0.3.6', type='build', when='care@:0.3.0')
     # 0.3.6 + specific fix for rocm
     depends_on('blt@0.3.6rocm', type='build', when='+rocm')

--- a/var/spack/repos/builtin/packages/care/package.py
+++ b/var/spack/repos/builtin/packages/care/package.py
@@ -29,7 +29,9 @@ class Care(CMakePackage, CudaPackage, ROCmPackage):
     variant('loop_fuser', default=False, description='Enable loop fusion capability')
 
     depends_on('blt', type='build')
-    depends_on('blt@0.3.7:', type='build', when='+rocm')
+    depends_on('blt@0.3.6', type='build', when='care@:0.3.0')
+    # 0.3.6 + specific fix for rocm
+    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
 
     depends_on('camp')
     depends_on('umpire@develop')

--- a/var/spack/repos/builtin/packages/care/package.py
+++ b/var/spack/repos/builtin/packages/care/package.py
@@ -28,10 +28,10 @@ class Care(CMakePackage, CudaPackage, ROCmPackage):
     variant('tests', default=False, description='Build tests')
     variant('loop_fuser', default=False, description='Enable loop fusion capability')
 
-    depends_on('blt', type='build', when='care@0.3.1:')
-    depends_on('blt@0.3.6', type='build', when='care@:0.3.0')
+    depends_on('blt', type='build', when='@0.3.1:')
+    depends_on('blt@:0.3.6', type='build', when='@:0.3.0')
     # 0.3.6 + specific fix for rocm
-    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
+    depends_on('blt@0.3.6rocm', type='build', when='@:0.3.0+rocm')
 
     depends_on('camp')
     depends_on('umpire@develop')

--- a/var/spack/repos/builtin/packages/chai/package.py
+++ b/var/spack/repos/builtin/packages/chai/package.py
@@ -35,8 +35,12 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
+
     depends_on('blt', type='build')
-    depends_on('blt@0.3.7:', type='build', when='+rocm')
+    depends_on('blt@0.3.6', type='build', when='chai@:2.3.0')
+    # 0.3.6 + specific fix for rocm
+    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
+
     depends_on('umpire')
     depends_on('raja', when="+raja")
 

--- a/var/spack/repos/builtin/packages/chai/package.py
+++ b/var/spack/repos/builtin/packages/chai/package.py
@@ -36,7 +36,7 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
 
-    depends_on('blt', type='build')
+    depends_on('blt', type='build', when='chai@2.3.1')
     depends_on('blt@0.3.6', type='build', when='chai@:2.3.0')
     # 0.3.6 + specific fix for rocm
     depends_on('blt@0.3.6rocm', type='build', when='+rocm')

--- a/var/spack/repos/builtin/packages/chai/package.py
+++ b/var/spack/repos/builtin/packages/chai/package.py
@@ -36,7 +36,7 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
 
-    depends_on('blt', type='build', when='chai@2.3.1')
+    depends_on('blt', type='build', when='chai@2.3.1:')
     depends_on('blt@0.3.6', type='build', when='chai@:2.3.0')
     # 0.3.6 + specific fix for rocm
     depends_on('blt@0.3.6rocm', type='build', when='+rocm')

--- a/var/spack/repos/builtin/packages/chai/package.py
+++ b/var/spack/repos/builtin/packages/chai/package.py
@@ -36,10 +36,10 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
 
-    depends_on('blt', type='build', when='chai@2.3.1:')
-    depends_on('blt@0.3.6', type='build', when='chai@:2.3.0')
+    depends_on('blt', type='build', when='@2.3.1:')
+    depends_on('blt@:0.3.6', type='build', when='@:2.3.0')
     # 0.3.6 + specific fix for rocm
-    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
+    depends_on('blt@0.3.6rocm', type='build', when='@:2.3.0+rocm')
 
     depends_on('umpire')
     depends_on('raja', when="+raja")

--- a/var/spack/repos/builtin/packages/raja/package.py
+++ b/var/spack/repos/builtin/packages/raja/package.py
@@ -37,10 +37,10 @@ class Raja(CMakePackage, CudaPackage, ROCmPackage):
     # and remove the +tests conflict below.
     variant('tests', default=False, description='Build tests')
 
-    depends_on('blt', type='build', when='raja@0.13.1:')
-    depends_on('blt@0.3.6', type='build', when='raja@:0.13.0')
+    depends_on('blt', type='build', when='@0.13.1:')
+    depends_on('blt@:0.3.6', type='build', when='@:0.13.0')
     # 0.3.6 + specific fix for rocm
-    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
+    depends_on('blt@0.3.6rocm', type='build', when='@:0.13.0+rocm')
 
     depends_on('camp')
     depends_on('camp+cuda', when='+cuda')

--- a/var/spack/repos/builtin/packages/raja/package.py
+++ b/var/spack/repos/builtin/packages/raja/package.py
@@ -38,7 +38,9 @@ class Raja(CMakePackage, CudaPackage, ROCmPackage):
     variant('tests', default=False, description='Build tests')
 
     depends_on('blt', type='build')
-    depends_on('blt@0.3.7:', type='build', when='+rocm')
+    depends_on('blt@0.3.6', type='build', when='raja@:0.13.0')
+    # 0.3.6 + specific fix for rocm
+    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
 
     depends_on('camp')
     depends_on('camp+cuda', when='+cuda')

--- a/var/spack/repos/builtin/packages/raja/package.py
+++ b/var/spack/repos/builtin/packages/raja/package.py
@@ -37,7 +37,7 @@ class Raja(CMakePackage, CudaPackage, ROCmPackage):
     # and remove the +tests conflict below.
     variant('tests', default=False, description='Build tests')
 
-    depends_on('blt', type='build')
+    depends_on('blt', type='build', when='raja@0.13.1:')
     depends_on('blt@0.3.6', type='build', when='raja@:0.13.0')
     # 0.3.6 + specific fix for rocm
     depends_on('blt@0.3.6rocm', type='build', when='+rocm')

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -60,7 +60,7 @@ class Umpire(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', when='+cuda', type='build')
 
-    depends_on('blt', type='build')
+    depends_on('blt', type='build', when='umpire@4.1.3:')
     depends_on('blt@0.3.6', type='build', when='umpire@:4.1.2')
     # 0.3.6 + specific fix for rocm
     depends_on('blt@0.3.6rocm', type='build', when='+rocm')

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -60,10 +60,10 @@ class Umpire(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', when='+cuda', type='build')
 
-    depends_on('blt', type='build', when='umpire@4.1.3:')
-    depends_on('blt@0.3.6', type='build', when='umpire@:4.1.2')
+    depends_on('blt', type='build', when='@4.1.3:')
+    depends_on('blt@:0.3.6', type='build', when='@:4.1.2')
     # 0.3.6 + specific fix for rocm
-    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
+    depends_on('blt@0.3.6rocm', type='build', when='@:4.1.2+rocm')
 
     # variants +rocm and amdgpu_targets are not automatically passed to
     # dependencies, so do it manually.

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -61,7 +61,9 @@ class Umpire(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.9:', when='+cuda', type='build')
 
     depends_on('blt', type='build')
-    depends_on('blt@0.3.7:', type='build', when='+rocm')
+    depends_on('blt@0.3.6', type='build', when='umpire@:4.1.2')
+    # 0.3.6 + specific fix for rocm
+    depends_on('blt@0.3.6rocm', type='build', when='+rocm')
 
     # variants +rocm and amdgpu_targets are not automatically passed to
     # dependencies, so do it manually.

